smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/integration/ProtocolGenerator.java

```diff
@@ -156,12 +157,18 @@ default void generateSharedComponents(GenerationContext context) {
     void generateFrameworkErrorSerializer(GenerationContext serverContext);
 
     /**
-     * Generates the code used to determine the service and operation
-     * targeted by a given request.
+     * Generates a factory for the ServiceHandler implementation for this service.
      *
      * @param context Generation context.
      */
-    void generateHandlerFactory(GenerationContext context);
+    void generateServiceHandlerFactory(GenerationContext context);
+
+    /**
+     * Generates the code used to handle a request for a specific operation in the given service.
+     *
+     * @param context Generation context.
+     */
```

Comment:
I think this could use some more context on why it exists. Also docs for the operation param